sdk/src/client.rs

```diff
@@ -131,33 +210,43 @@ impl NetworkClient {
         Ok(res.proof_id)
     }
 
-    pub async fn get_proof_status<SC: StarkGenericConfig + Serialize + DeserializeOwned>(
-        &self,
-        proof_id: &str,
-    ) -> Result<(GetProofStatusResponse, Option<SP1ProofWithIO<SC>>)> {
+    // Claim a proof that was requested. This commits to generating a proof and fulfilling it.
+    pub async fn claim_proof(&self, proof_id: &str) -> Result<ClaimProofResponse> {
+        let nonce = self.get_nonce().await;
+        let signature = self.auth.sign_claim_proof_message(nonce, proof_id).await?;
         let res = self
             .rpc
-            .get_proof_status(GetProofStatusRequest {
+            .claim_proof(ClaimProofRequest {
+                nonce,
                 proof_id: proof_id.to_string(),
+                signature,
             })
             .await?;
 
-        let proof = if res.status() == ProofStatus::ProofSucceeded {
-            let proof = self
-                .http
-                .get(res.result_get_url.clone())
-                .send()
-                .await?
-                .bytes()
-                .await?;
-            Some(bincode::deserialize(&proof).expect("Failed to deserialize proof"))
-        } else {
-            None
-        };
+        Ok(res)
+    }
 
-        Ok((res, proof))
+    // Fulfill a proof. Should only be called after the proof has been uploaded. Only callable by the
+    // prover that claimed the proof.
+    pub async fn fulfill_proof(&self, proof_id: &str) -> Result<FulfillProofResponse> {
+        let nonce = self.get_nonce().await;
+        let signature = self
+            .auth
+            .sign_fulfill_proof_message(nonce, proof_id)
+            .await?;
+        let res = self
+            .rpc
+            .fulfill_proof(FulfillProofRequest {
+                nonce,
+                proof_id: proof_id.to_string(),
+                signature,
+            })
+            .await?;
+
+        Ok(res)
     }
 
+    // Relay a proof. Should only be called after a proof has been fulfilled.
```

Comment:
nit: Relay a proof. Returns an error if the proof has not been fulfilled yet.

Reply:
added